### PR TITLE
[feature-wip](Cloud) Support set default storage vault and pass vault id to all write paths

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Issue Number: close #xxx
1. Support set default storage vault. All tables which don't specify storage vault name would use this vault by default. **Notice** if there is no default vault and the table doesn't specify it's vault then it would fail.
2. Pass the vault id to all write path.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: gensrc/proto/cloud.proto [hunk 1]
```diff
@@ -1379,6 +1398,8 @@ service MetaService {
 
     // routine load progress
     rpc get_rl_task_commit_attach(GetRLTaskCommitAttachRequest) returns (GetRLTaskCommitAttachResponse);
+
+    rpc get_default_vault(GetDefaultVaultRequest) returns (GetDefaultVaultResponse);
```
Comment (gavinchou) [inline gensrc/proto/cloud.proto:1402]: no need this rpc

File: cloud/src/meta-service/meta_service.cpp [hunk 2]
```diff
@@ -1885,6 +1885,57 @@ void MetaServiceImpl::get_delete_bitmap_update_lock(google::protobuf::RpcControl
     }
 }
 
+void MetaServiceImpl::get_default_vault(::google::protobuf::RpcController* controller,
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service.cpp:1888]: no need this rpc

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 3]
```diff
@@ -399,13 +408,22 @@ void MetaServiceImpl::alter_obj_store_info(google::protobuf::RpcController* cont
         };
     } break;
     case AlterObjStoreInfoRequest::ADD_HDFS_INFO:
+    case AlterObjStoreInfoRequest::ADD_BUILT_IN_HDFS_INFO:
```
[the remark this answers is not in the document]
Comment (ByteYue) [reply cloud/src/meta-service/meta_service_resource.cpp:411]: It's not limited to HDFS, it's just one naming problem...

[the remark this answers is not in the document]
Comment (ByteYue) [reply cloud/src/meta-service/meta_service_resource.cpp:411]: 创造built-in vault的逻辑对于hdfs和s3都是一样的  这个enum我改一下

**Comment by ByteYue** [comment]
run buildall

File: be/src/cloud/cloud_storage_engine.cpp [hunk 4]
```diff
@@ -136,9 +136,7 @@ Status CloudStorageEngine::open() {
 
         LOG(WARNING) << "failed to get vault info, retry after 5s, err=" << st;
         std::this_thread::sleep_for(5s);
-    } while (true);
-
-    CHECK(!vault_infos.empty()) << "no vault infos";
+    } while (vault_infos.empty());
```
Comment (gavinchou) [inline be/src/cloud/cloud_storage_engine.cpp:139]: while true and remember to clear vault_info each iteration

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**